micrometer-observation-test/src/main/java/io/micrometer/observation/tck/ObservationContextAssert.java

```diff
@@ -275,28 +275,20 @@ public SELF doesNotHaveMapEntry(Object key, Object value) {
     }
 
     public SELF doesNotHaveError() {
-        isNotNull();
-        Optional<Throwable> error = this.actual.getError();
-        error.ifPresent(throwable -> failWithMessage("Observation should not have an error, found <%s>", throwable));
+        thenError().as("Observation should not have an error, found <%s>", this.actual.getError().orElse(null))
```

Comment:
how about using `withFailMessage` instead of `as` (which puts a prefix to the failure message but still outputs additional info from the `isNull` assertion that isn't terribly interesting)?

Reply:
to illustrate, with `as` it leads to a message similar to:
```
[Observation should not have an error, found <java.lang.IllegalStateException: test>] expected: 
  null
 but was: 
  java.lang.IllegalStateException: test
  	at io.micrometer.observation.tck.ObservationContextAssertTests.should_throw_when_unexpected_error(ObservationContextAssertTests.java:283)
  	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
  	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
  	...(67 remaining lines not displayed - this can be changed with Assertions.setMaxStackTraceElementsDisplayed)"
```